src/test/java/cloud/fogbow/ras/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -1,942 +1,1254 @@
-
+    public void testGetInstanceFailWithPublicIpOrder() throws FogbowException {
+        // set up
+        Order order = createLocalPublicIpOrder(FAKE_COMPUTE_ID);
+        order.setInstanceId(FAKE_INSTANCE_ID);
+        order.setOrderState(OrderState.FULFILLED);
+
+        CloudUser cloudUser = Mockito.mock(CloudUser.class);
+        Mockito.when(this.mapperPlugin.map(Mockito.any(SystemUser.class))).thenReturn(cloudUser);
+
+        String expected = Messages.Exception.NULL_VALUE_RETURNED;
+
+        try {
+            // exercise
+            this.localCloudConnector.getInstance(order);
+            Assert.fail();
+        } catch (UnexpectedException e) {
+            // verify
+            Assert.assertEquals(expected, e.getMessage());
+            Mockito.verify(this.localCloudConnector, Mockito.times(BaseUnitTests.RUN_ONCE)).auditRequest(
+                    Mockito.eq(Operation.GET), Mockito.eq(ResourceType.PUBLIC_IP), Mockito.any(SystemUser.class),
+                    Mockito.anyString());
+        }
+    }
+    
+    // test case: When invoking the getInstance method with a compute order with
+    // instance ID null it must call the createEmptyInstance method, and confirm in
+    // auditRequest the GET operation and the COMPUTE resource type.
+    @Test
+    public void testGetInstanceEmptyForComputeOrder() throws FogbowException {
         // set up
-        this.order = Mockito.mock(VolumeOrder.class);
-        Mockito.when(this.order.getType()).thenReturn(ResourceType.VOLUME);
-        Mockito.when(this.order.getInstanceId()).thenReturn(null);
-        Mockito.when(this.order.getId()).thenReturn(FAKE_ORDER_ID);
-        Mockito.when(this.order.getOrderState()).thenReturn(OrderState.OPEN);
+        Order order = createLocalComputeOrder();
+        order.setOrderState(OrderState.FULFILLED);
+
+        CloudUser cloudUser = Mockito.mock(CloudUser.class);
+        Mockito.when(this.mapperPlugin.map(Mockito.any(SystemUser.class))).thenReturn(cloudUser);
 
         // exercise
-        OrderInstance instance = this.localCloudConnector.getInstance(order);
+        this.localCloudConnector.getInstance(order);
 
-        //verify
-        Assert.assertEquals(FAKE_ORDER_ID, instance.getId());
-        Assert.assertEquals(InstanceState.DISPATCHED, instance.getState());
-        Mockito.verify(computePlugin, times(0)).getInstance(Mockito.any(ComputeOrder.class), Mockito.any(CloudUser.class));
-        Mockito.verify(volumePlugin, times(0)).getInstance(Mockito.any(VolumeOrder.class), Mockito.any(CloudUser.class));
-        Mockito.verify(attachmentPlugin, times(0)).getInstance(Mockito.any(AttachmentOrder.class), Mockito.any(CloudUser.class));
-        Mockito.verify(networkPlugin, times(0)).getInstance(Mockito.any(NetworkOrder.class), Mockito.any(CloudUser.class));
+        // verify
+        Mockito.verify(this.localCloudConnector, Mockito.times(BaseUnitTests.RUN_ONCE))
+                .createEmptyInstance(Mockito.eq(order));
```

Comment:
this test must be done when you're testing doGetInstance. Here, we just wanna know if doGetInstance has been called.

Reply:
same thing for the others GetInstanceEmpty tests

Reply:
take a look in the other tests to check if you're not doing the same thing there please

Reply:
Okay, understood ... I'll fix them.